test/parallel/test-fs-truncate.js

```diff
@@ -114,3 +113,51 @@ function testFtruncate(cb) {
     });
   });
 }
+
+
+// make sure if the size of the file is smaller than the length then it is
+// filled with zeroes.
+const filename1 = path.resolve(tmp, 'truncate-file-1.txt');
+
+{
+  fs.writeFileSync(filename1, 'Hi');
+  fs.truncateSync(filename1, 4);
+  assert.strictEqual(
+    fs.readFileSync(filename1).compare(Buffer.from('Hi\u0000\u0000')), 0
```

Comment:
Using `assert.ok` + `.equals` instead of `.compare` should get this under 80 characters